tests/foreman/maintain/test_service.py

```diff
@@ -0,0 +1,267 @@
+"""Test class for satellite-maintain service functionality
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ForemanMaintain
+
+:Assignee: gtalreja
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+
+from robottelo.config import settings
+from robottelo.constants import HAMMER_CONFIG
+from robottelo.constants import maintain_hammer_yml
+from robottelo.constants import satellite_answer_file
+from robottelo.utils.issue_handlers import is_open
+
+pytestmark = pytest.mark.destructive
+
+
+@pytest.fixture
+def missing_hammer_config(request, sat_maintain):
+    """Setup/teardown fixture used by test_positive_service_restart_without_hammer_config"""
+    flag = 'initial_admin_password'
+    sat_maintain.execute(
+        f'sed -i "s/{flag}: {settings.server.admin_password}/{flag}: invalid/g" '
+        f'{satellite_answer_file}'
+    )
+    sat_maintain.execute(f'mv {HAMMER_CONFIG} /tmp/foreman.yml')
+    sat_maintain.execute(f'mv {maintain_hammer_yml} /tmp/foreman-maintain-hammer.yml')
+
+    @request.addfinalizer
+    def _finalize():
+        sat_maintain.execute(
+            f'sed -i "s/{flag}: invalid/{flag}: {settings.server.admin_password}/g" '
+            f'{satellite_answer_file}'
+        )
+        sat_maintain.execute(f'mv /tmp/foreman.yml {HAMMER_CONFIG}')
+        sat_maintain.execute(f'mv /tmp/foreman-maintain-hammer.yml {maintain_hammer_yml}')
+
+
+@pytest.mark.include_capsule
+def test_positive_service_list(sat_maintain):
+    """List satellite services using satellite-maintain service subcommand
+
+    :id: aa878d16-5a22-11ed-849d-cb7ac12cece2
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service list
+
+    :expectedresults: Satellite services should be listed
+    """
+    result = sat_maintain.cli.Service.list()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_service_stop_start(sat_maintain):
+    """Start/Stop services using satellite-maintain service subcommand
+
+    :id: fa3b02a9-b441-413b-b9d2-1a59d04f285c
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service stop
+        2. Run satellite-maintain service status --brief
+        3. Run satellite-maintain service start
+        4. Run satellite-maintain service status --brief
+
+    :expectedresults: Satellite services should stop and start.
+    """
+    result = sat_maintain.cli.Service.stop()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' in result.stdout
```

Comment:
Can you add a comment above as to what actually gets into FAIL state ?
OR you could you be specific as `FAIL something> string so that we will not false pass this test in actual FAIL message for service stopping.

Reply:
Steps are quite straight forward here, we're stopping all services first then check status for all services

Reply:
Read my comment again :)

Reply:
@jyejare It's hard to tell why it could get in fail state and that's for all the foreman-maintain commands.
For this one, mostly it'd fail if a service is not running.
FM shows FAIL/OK for check/command we run and that's what we really care about. 
And if command/check doesn't fail then usually there's no unnecessary o/p so it's very unlikely that we'll get `FAIL` in output and that test will false pass.

```
| displaying dynflow-sidekiq@orchestrator          [FAIL]                       
| displaying foreman                               [FAIL]                       
| displaying httpd                                 [OK]
```

Reply:
@jameerpathan111 That clears everything !

Reply:
Thank you @jameerpathan111 @jyejare